File: tests/hello.cpp
#!/usr/bin/env jumpscript -S C++
# nix: {
#   buildInputs = [ pkgs.boost ];
# }

#include <iostream>
#include <vector>
#include <string>
#include <boost/algorithm/string.hpp>

int main(int argc, char *argv[]) {
    std::cout << "Hello from C++!\n";
    
    // Process arguments
    std::vector<std::string> args(argv + 1, argv + argc);
    
    if (!args.empty()) {
        std::cout << "Arguments:\n";
        for (size_t i = 0; i < args.size(); ++i) {
            std::cout << "  " << i+1 << ": " << args[i] << "\n";
        }
        
        // Demonstrate using boost
        std::string joined = boost::algorithm::join(args, ", ");
        std::cout << "All args joined: " << joined << "\n";
    }
    
    return 0;
}

File: tests/hello.zig
#!/usr/bin/env jumpscript -S Zig
# nix: {
#   buildInputs = [];
# }

const std = @import("std");

pub fn main() !void {
    const stdout = std.io.getStdOut().writer();
    try stdout.print("Hello from Zig!\n", .{});
    
    // Process command line arguments
    var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
    defer arena.deinit();
    const allocator = arena.allocator();
    
    const args = try std.process.argsAlloc(allocator);
    defer std.process.argsFree(allocator, args);
    
    if (args.len > 1) {
        try stdout.print("Arguments:\n", .{});
        for (args[1..], 0..) |arg, i| {
            try stdout.print("  {d}: {s}\n", .{ i + 1, arg });
        }
        
        // Try to parse numeric arguments and calculate their sum
        var sum: i64 = 0;
        var has_numeric = false;
        
        for (args[1..]) |arg| {
            const parsed = std.fmt.parseInt(i64, arg, 10) catch continue;
            sum += parsed;
            has_numeric = true;
        }
        
        if (has_numeric) {
            try stdout.print("Sum of numeric arguments: {d}\n", .{sum});
        }
    }
}
